src/frontend/test/features/pageobjects/fillPPQ.page.js

```diff
@@ -1,83 +1,81 @@
 const Page = require('./page');
 const link = "//a[contains(text(),'";
 const randomLink = "(//label[contains(text(),'')])[";
-
+const mpoCheckboxLabel =
+  'I acknowledge that I have participated in the drafting and/or review of this PIA on behalf of my ministry or sector. All privacy risks of which I am aware are documented with acceptable risk mitigations in place and I have no outstanding privacy concerns related to the initiative under review.';
 
 /**
  * sub page containing specific selectors and methods for a specific page
  */
 class fillPPQ extends Page {
-    /**
-     * define selectors using getter methods
-     */
-    get datePicker () {
-      return $('//input[@placeholder="yyyy/mm/dd"]');
-   }
+  /**
+   * define selectors using getter methods
+   */
+  get datePicker() {
+    return $('//input[@placeholder="yyyy/mm/dd"]');
+  }
 
-   get selectDate () {
+  get selectDate() {
     return $("//div[contains(text(),'10')]");
-   }
-
-   get textArea () {
-    return $("//textarea");
-   }
-
-    
+  }
 
-    /**
-     * a method to encapsule automation code to interact with the page
-     * e.g. to login using username and password
-     */
+  get textArea() {
+    return $('//textarea');
+  }
 
+  /**
+   * a method to encapsule automation code to interact with the page
+   * e.g. to login using username and password
+   */
 
-  
-    
-    async selectCheckbox (checkName) {
-        const h1=link + checkName + "')]/../input";
-        const headerElement=await $(h1);
-        await headerElement.waitForDisplayed();
+  async selectCheckbox(checkName) {
```

Comment:
Im confused how this works in the else block and whats going on between selectCheckbox and clickCheckbox

Reply:
> Im confused how this works in the else block and whats going on between selectCheckbox and clickCheckbox

common function method like button, href, etc or selectors are managed under the page.js file so as not to duplicate the code and other sub pages extends the page.js to use the common methods. selectCheckbox function to handle the parameter of the checkbox name when we check the specific checkbox step under the feature file but the clickCheckbox function is for the random checkbox scenarios(generating a random number and selecting any random input checkbox).